### PR TITLE
Rwahs 72/some duplicate null values being added to the donors list

### PR DESCRIPTION
**Pull request by kehh**
- Refactor import to a series of functions in an include file
  - Import scripts now simply define a number of parameters and then call import
- Add an export view for Sources, remove source information from ObjectsExportView
- Run donors import first - as this is the first import, it has to load the data out of the access database
- Remove additional donor information from object import ￼…
  - Sample data for donor import
  - Update sample records for `adminDonor` as this no longer includes donor information
  - Fix quoting for called function importData
  - Update to donor and museum import mapping

### CODE REVIEW

File: museum/sampleRecords/donors.destination.json [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+  "ok":true,
+  "results":[
+    {
+      "entity_id":"1",
+      "id":"1",
+      "idno":"1",
+      "display_label":"N\/A",
+      "ca_entities.idno":"1",
+      "ca_entities.preferred_labels.displayname":"N\/A",
+      "ca_entities.preferred_labels.surname":"&lt;&gt;"
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json]: This doesn't look right, I think `surname` should be empty.

File: museum/sampleRecords/donors.destination.json [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+    },
+    {
+      "entity_id":"5",
+      "id":"5",
+      "idno":"6",
+      "display_label":"FAIRBAIRN, Miss, and Dr Fairbairn",
+      "ca_entities.idno":"6",
+      "ca_entities.preferred_labels.displayname":"FAIRBAIRN, Miss, and Dr Fairbairn",
+      "ca_entities.preferred_labels.prefix":"Miss",
+      "ca_entities.preferred_labels.forename":"Miss, and Dr Fairbairn",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:71]: This doesn't look right, although I'm not sure what can be done about it.  It should be `surname: "Fairbairn", prefix "Miss and Dr"`, or possibly two separate Donor records, although I'm not sure how the mapping will generate either of those.  Thoughts?

File: museum/sampleRecords/donors.destination.json [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+    {
+      "entity_id":"2",
+      "id":"2",
+      "idno":"2",
+      "display_label":"HOPE A.T.",
+      "ca_entities.idno":"2",
+      "ca_entities.preferred_labels.displayname":"HOPE A.T.",
+      "ca_entities.preferred_labels.prefix":"Mrs",
+      "ca_entities.preferred_labels.forename":"A.T.",
+      "ca_entities.preferred_labels.middlename":"A.T.",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:22]: I can't remember what we discussed at the meeting but `forename` and `middlename` are duplicated.  Is that a separate task to fix that issue?

Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:22]: Actually comparing to `RILEY` below I can see this duplication comes from the source data, so it's probably okay?

File: museum/sampleRecords/donors.destination.json [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+      "ca_entities.LastEditBy":"IST",
+      "ca_entities.LastEditDate":"19 December 2000"
+    },
+    {
+      "entity_id":"8",
+      "id":"8",
+      "idno":"9",
+      "display_label":"ANDERSON Nell",
+      "ca_entities.idno":"9",
+      "ca_entities.preferred_labels.displayname":"ANDERSON Nell",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:118]: Shouldn't this be `ANDERSON, Nell`?  (with comma)

File: museum/sampleRecords/donors.destination.json [hunk 5]
```diff
@@ -0,0 +1,578 @@
...
+      "ca_entities.LastEditBy":"JILLM",
+      "ca_entities.LastEditDate":"16 March 2004"
+    },
+    {
+      "entity_id":"9",
+      "id":"9",
+      "idno":"10",
+      "display_label":"KIRWAN M",
+      "ca_entities.idno":"10",
+      "ca_entities.preferred_labels.displayname":"KIRWAN M",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:135]: Same here, shouldn't this be `KIRWAN, M`?  Compare `CLIFTON, Edmund` above.

File: museum/sampleRecords/donors.destination.json [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+      "entity_id":"11",
+      "id":"11",
+      "idno":"12",
+      "display_label":"LLOYD, MRS Stephanie",
+      "ca_entities.idno":"12",
+      "ca_entities.preferred_labels.displayname":"LLOYD, MRS Stephanie",
+      "ca_entities.preferred_labels.prefix":"Mrs",
+      "ca_entities.preferred_labels.forename":"Stephanie",
+      "ca_entities.preferred_labels.middlename":"S.",
+      "ca_entities.preferred_labels.surname":"Lloyd, Mrs",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:176]: This doesn't look right, `surname` should just be `Lloyd`, `Mrs` only belongs in `prefix`.

File: museum/sampleRecords/donors.destination.json [hunk 7]
```diff
@@ -0,0 +1,578 @@
...
+      "ca_entities.AddressContainer.Address1":"1c Woodhouse Street, East Fremantle, 6158",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:177]: This shouldn't have the suburb and postcode, but I can see that comes from the source data.  Something for the client to clean.

File: museum/sampleRecords/donors.destination.json [hunk 8]
```diff
@@ -0,0 +1,578 @@
...
+    },
+    {
+      "entity_id":"13",
+      "id":"13",
+      "idno":"29",
+      "display_label":"RWAHS",
+      "ca_entities.idno":"29",
+      "ca_entities.preferred_labels.displayname":"RWAHS",
+      "ca_entities.preferred_labels.surname":"RWAHS",
+      "ca_entities.AddressContainer.Address1":"49 Broadway, Nedlands",
```
Comment (leftclickben) [inline museum/sampleRecords/donors.destination.json:206]: More dirty source data.

**Comment by leftclickben** [comment]
Let's discuss tomorrow, possibly nothing to change but a few points to go through.

File: museum/sampleRecords/donors.destination.json (hunk 1 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json]: This appears to be an artefact of the web services: http://staging-collections.histwest.org.au/index.php/editor/entities/EntityEditor/Edit/entity_id/1

File: museum/sampleRecords/donors.destination.json (hunk 3 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json:22]: Yes it's a separate issue, and yes it's content which will be cleaned up.

File: museum/sampleRecords/donors.destination.json (hunk 2 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json:71]: This is the source data, and it is possible to change the entity type of this entity to a 'FAMILY' which then has simpler labels, or add an additional name to the entity, or have two entities. but I think just keep it as they have been using it.

File: museum/sampleRecords/donors.destination.json (hunk 4 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json:118]: This should be fixed in the source data. Alternatively we could consistently format the display name as you suggest, but I think it's better to leave it as they have it in the source.

File: museum/sampleRecords/donors.destination.json (hunk 6 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json:176]: cleaning

File: museum/sampleRecords/donors.destination.json (hunk 7 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json:177]: cleaning

File: museum/sampleRecords/donors.destination.json (hunk 1 above)
Comment (kehh) [inline museum/sampleRecords/donors.destination.json]: Actually I had fixed this, just not run the test again since fixing it.

Comment (kehh) [inline museum/sampleRecords/donors.destination.json]: source data has surname as `<<N/A>>` - this is transformed below to `N/A`.

**Pull request merged**